File: jiWoo/프로그래머스_카펫.java
package jiWoo;

public class 프로그래머스_카펫 {
    class Solution {
        public int[] solution(int brown, int yellow) {
            int x=3; // x는 세로 값 y를 가로 값으로 했을 때 2y+2x-4= brown, (x-2))y-2)=yellow로 식을 구해서 다 돌려서 탐색함!
            int val=brown+yellow;
            int[] answer=new int[2];
            while(true){
                if(val%x==0 && val/x>=x && (2*x+2*(val/x)-4)==brown){
                    answer[1]=x;
                    answer[0]=val/x;
                    break;
                }
                x++;
            }
            return answer;
        }
    }
}

File: yunhwan/프로그래머스 카펫.py
def solution(brown, yellow):
    answer = []

		#총 격자의 수
    total = brown + yellow
    row = 1
    col = total//row
    
	
    for i in range(1,brown):
        # row * col =total 이용
				row = i
        col = total//row
				#조건을 나열하여 만족한다면 break
				#노란색 타일은 row-2 * col-2
				#row >= col
				#갈색 타일은 2*row + 2*col -4(중복타일 제거)
        if (row-2) * (col-2) == yellow and row>=col and 2*row +2*col -4 ==brown:
            break
                
    answer = [row,col]
    
    return answer